controlplane/kubeadm/internal/cluster.go

```diff
@@ -70,17 +87,25 @@ func OwnedControlPlaneMachines(controlPlaneName string) func(machine *clusterv1.
 // that have a deletion timestamp.
 func HasDeletionTimestamp() func(machine *clusterv1.Machine) bool {
 	return func(machine *clusterv1.Machine) bool {
-		return machine.GetDeletionTimestamp() != nil
+		return !WithoutDeletionTimestamp()(machine)
 	}
 }
 
-// HasOutdatedConfiguration returns a MachineFilter function to find all machines
-// that do not match a given KubeadmControlPlane configuration hash.
-func HasOutdatedConfiguration(configHash string) func(machine *clusterv1.Machine) bool {
+// WithoutDeletionTimestamp returns a MachineFilter function to find all machines
+// that do not have a deletion timestamp.
+func WithoutDeletionTimestamp() func(machine *clusterv1.Machine) bool {
 	return func(machine *clusterv1.Machine) bool {
 		if machine == nil {
 			return false
 		}
+		return machine.DeletionTimestamp.IsZero()
+	}
+}
+
+// HasOutdatedConfiguration returns a MachineFilter function to find all machines
+// that do not match a given KubeadmControlPlane configuration hash.
+func HasOutdatedConfiguration(configHash string) func(machine *clusterv1.Machine) bool {
```

Comment:
Not for this PR (unless you want to), but we should think about extracting these into machine_filters.go or similar file.